modules/porous_flow/src/materials/PorousFlowEffectiveFluidPressure.C

```diff
@@ -0,0 +1,59 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowEffectiveFluidPressure.h"
+
+template<>
+InputParameters validParams<PorousFlowEffectiveFluidPressure>()
+{
+  InputParameters params = validParams<Material>();
+
+  params.addRequiredParam<UserObjectName>("PorousFlowDictator_UO", "The UserObject that holds the list of Porous-Flow variable names.");
+  params.addClassDescription("This Material calculates an effective fluid pressure: effective_stress = total_stress + biot_coeff*effective_fluid_pressure.  The effective_fluid_pressure = sum_{phases}(S_phase * P_phase)");
+  return params;
+}
+
+PorousFlowEffectiveFluidPressure::PorousFlowEffectiveFluidPressure(const InputParameters & parameters) :
+    DerivativeMaterialInterface<Material>(parameters),
+
+    _dictator_UO(getUserObject<PorousFlowDictator>("PorousFlowDictator_UO")),
+    _num_ph(_dictator_UO.numPhases()),
+    _num_var(_dictator_UO.numVariables()),
+
+    _porepressure_qp(getMaterialProperty<std::vector<Real> >("PorousFlow_porepressure_qp")),
+    _dporepressure_qp_dvar(getMaterialProperty<std::vector<std::vector<Real> > >("dPorousFlow_porepressure_qp_dvar")),
+    _saturation_qp(getMaterialProperty<std::vector<Real> >("PorousFlow_saturation_qp")),
+    _dsaturation_qp_dvar(getMaterialProperty<std::vector<std::vector<Real> > >("dPorousFlow_saturation_qp_dvar")),
+    _porepressure_nodal(getMaterialProperty<std::vector<Real> >("PorousFlow_porepressure_nodal")),
+    _dporepressure_nodal_dvar(getMaterialProperty<std::vector<std::vector<Real> > >("dPorousFlow_porepressure_nodal_dvar")),
+    _saturation_nodal(getMaterialProperty<std::vector<Real> >("PorousFlow_saturation_nodal")),
+    _dsaturation_nodal_dvar(getMaterialProperty<std::vector<std::vector<Real> > >("dPorousFlow_saturation_nodal_dvar")),
+    _pf_qp(declareProperty<Real>("PorousFlow_effective_fluid_pressure_qp")),
+    _dpf_qp_dvar(declareProperty<std::vector<Real> >("dPorousFlow_effective_fluid_pressure_qp_dvar")),
+    _pf_nodal(declareProperty<Real>("PorousFlow_effective_fluid_pressure_nodal")),
+    _dpf_nodal_dvar(declareProperty<std::vector<Real> >("dPorousFlow_effective_fluid_pressure_nodal_dvar"))
+{
+}
+
+void
+PorousFlowEffectiveFluidPressure::computeQpProperties()
+{
+  _pf_qp[_qp] = 0;
+  _pf_nodal[_qp] = 0;
```

Comment:
I'd prefer `0.0` over `0` if those are floating point variables. Makes the code clearer IMO.